File: lib/common/MenuItem.dart
import 'dart:io';

import 'package:flutter/material.dart';

// Define the class
class MenuItem {
  int? kitchenId;
  int? itemId;
  String? image;
  String? name;
  String? notes;
  int? quantity;
  int? category;
  double? price;
  String? time;
  String? cName;
 
  MenuItem({
    this.kitchenId,
    this.itemId,
    this.image,
    this.name,
    this.notes,
    this.quantity,
    this.category,
    this.price,
    this.time,
    this.cName = "",
  });
}

File: server/routes/menu item/menu_item.js
const express = require("express");
const router = express.Router();
const pool = require("../../db");

router.post("/", async (req, res) => {
  const {
    name,
    image,
    notes,
    kitchen_id,
    category_id,
    quantity_id,
    price,
    time
  } = req.body;

  const query = `
  INSERT INTO menu_item (name, image, notes, kitchen_id, category_id, quantity_id, price, time) 
  VALUES (?, ?, ?, ?, ?, ?, ?, ?)
  `;

  pool.execute(
    query,
    [name, image, notes, kitchen_id, category_id, quantity_id, price, time],
    (error, results, fields) => {
      if (error) {
        console.error("Error inserting values:", error);
        res.status(500).send("Error inserting values");
        return;
      }
      res.status(200).send("A new menu item inserted successfully!");
    }
  );
});

module.exports = router;
